htlcswitch/failure_test.go

```diff
@@ -0,0 +1,82 @@
+package htlcswitch
+
+import (
+	"encoding/hex"
+	"testing"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	sphinx "github.com/lightningnetwork/lightning-onion"
+	"github.com/lightningnetwork/lnd/lnwire"
+	"github.com/lightningnetwork/lnd/tlv"
+	"github.com/stretchr/testify/require"
+)
+
+// TestLongFailureMessage tests that longer failure messages can be interpreted
+// correctly.
+func TestLongFailureMessage(t *testing.T) {
+	t.Parallel()
+
+	// Use long 1024-byte test vector from BOLT 04.
+	sessionKeyBytes, _ := hex.DecodeString("4141414141414141414141414141414141414141414141414141414141414141") //nolint:lll
+
+	pathKeys := []string{
+		"02eec7245d6b7d2ccb30380bfbe2a3648cd7a942653f5aa340edcea1f283686619", //nolint:lll
```

Comment:
nit: can use the `//notlint:lll` for the whole test function to avoid needing to append it to each already long line.

Reply:
Hmm yes, but then it wouldn't be applied to the code in this function.

Reply:
I mean like this:
```
// TestLongFailureMessage tests that longer failure messages can be interpreted
// correctly.
//
//nolint:lll
func TestLongFailureMessage(t *testing.T) {
```

Then the comment is valid for the whole block it precedes.

Reply:
Yes, got it. But then the code in that function is also excluded from lll?

Reply:
Ah, I guess so, yes.

Reply:
Moved all of it into a json file.